src/java.base/unix/classes/sun/security/provider/NativePRNG.java

```diff
@@ -536,6 +614,15 @@ private void ensureBufferValid() throws IOException {
         private void implNextBytes(byte[] data) {
                 try {
                     getMixRandom().engineNextBytes(data);
+
+                    /*[IF CRIU_SUPPORT]*/
+                    // read random data from non blocking source
+                    byte[] rawData = new byte[data.length];
+                    readFully(nextIn, rawData);
+                    for (int i = 0; i < data.length; i++) {
+                        data[i] ^= rawData[i];
+                    }
```

Comment:
What is this for and why is it not applied only for `Variant.CRIU`?

Reply:
> It is only applied for Variant.CRIU because it is surrounded by the /_[IF CRIU_SUPPORT]_/ condition

No, `/*[IF CRIU_SUPPORT]*/` means the code is included in builds where CRIU support is _available_. That doesn't mean that the object is related to `Variant.CRIU` due to the presence of `-XX:+EnableCRIUSupport` on the command line.

Reply:
Updated, adding a new `flag` for this, so it can be applied only for `Variant.CRIU`.